nibabel/freesurfer/io.py

```diff
@@ -400,6 +447,13 @@ def write_string(s):
             write(len(s))
             write(s, dtype='|S%d' % len(s))
 
+        # Generate annotation values for each ctab entry
+        if fill_ctab:
+            ctab = np.hstack((ctab[:, :4], _pack_rgba(ctab[:, :4])))
+        elif not np.array_equal(ctab[:, 4], _pack_rgba(ctab[:, :4])):
```

Comment:
The shape is off here, so it's always going to evaluate `True`. Your left entry needs the square brackets around the 4. Maybe add a test with a valid and invalid annotation values, to make sure the warning is raised only in the latter case.

Reply:
Yep, the warning was always being raised. Fixed, and covered in the tests.